libff/algebra/curves/bn128/bn128_g1.cpp

```diff
@@ -55,7 +57,7 @@ bn::Fp bn128_G1::sqrt(const bn::Fp &el)
             m += 1;
         }
 
-        int j = v-m-1;
+        size_t j = v-m-1;
```

Comment:
Hum... this type change looks fishy. `size_t` is unsigned (see: https://en.cppreference.com/w/cpp/types/size_t) which means that j will always be >= 0. Luckily this does not create an infinite loop (because we exit when j reaches 0) but, the logic is altered (especially if `v-m-1` is negative and effectively "wraps around" during the initialization of `j` before the `while`).

Reply:
Thanks for catching this!

Reply:
Thanks for catching this! I've updated it, and didn't catch any other occurences of a similar implicit signed to unsigned converison